diff -urp tpm-tools-1.3.8.orig/lib/Makefile.am tpm-tools-1.3.8/lib/Makefile.am
--- tpm-tools-1.3.8.orig/lib/Makefile.am	2012-05-17 13:50:39.000000000 -0400
+++ tpm-tools-1.3.8/lib/Makefile.am	2012-06-21 13:07:29.624445940 -0400
@@ -52,7 +52,7 @@ if P11_SUPPORT
 noinst_LTLIBRARIES	+=	libtpm_pkcs11.la
 
 libtpm_pkcs11_la_SOURCES=	tpm_pkcs11.c
-libtpm_pkcs11_la_LIBADD	=	libtpm_utils.la
+libtpm_pkcs11_la_LIBADD	=	libtpm_utils.la -ldl
 endif
 
 #
diff -urp tpm-tools-1.3.8.orig/lib/tpm_tspi.c tpm-tools-1.3.8/lib/tpm_tspi.c
--- tpm-tools-1.3.8.orig/lib/tpm_tspi.c	2011-08-17 08:20:35.000000000 -0400
+++ tpm-tools-1.3.8/lib/tpm_tspi.c	2012-06-21 13:07:29.654445942 -0400
@@ -702,14 +702,14 @@ TSS_RESULT
 unloadNVDataPublic(UINT64 *offset, BYTE *blob, UINT32 blob_len, TPM_NV_DATA_PUBLIC *v)
 {
 	UINT64 off = *offset;
-	TSS_RESULT result;
-	result = Trspi_UnloadBlob_NV_DATA_PUBLIC(&off, blob, NULL);
+	TSS_RESULT result = TSS_SUCCESS;
+/*	result = Trspi_UnloadBlob_NV_DATA_PUBLIC(&off, blob, NULL);
 	if (result == TSS_SUCCESS) {
 		if (off > blob_len)
 			return TSS_E_BAD_PARAMETER;
 		result = Trspi_UnloadBlob_NV_DATA_PUBLIC(offset, blob, v);
 	}
-	tspiResult("Trspi_UnloadBlob_NV_DATA_PUBLIC", result);
+	tspiResult("Trspi_UnloadBlob_NV_DATA_PUBLIC", result); */
 	return result;
 }
 
diff -urp tpm-tools-1.3.8.orig/src/tpm_mgmt/Makefile.am tpm-tools-1.3.8/src/tpm_mgmt/Makefile.am
--- tpm-tools-1.3.8.orig/src/tpm_mgmt/Makefile.am	2012-05-17 13:50:39.000000000 -0400
+++ tpm-tools-1.3.8/src/tpm_mgmt/Makefile.am	2012-06-21 13:07:29.655445942 -0400
@@ -23,6 +23,8 @@
 
 noinst_HEADERS	=	tpm_nvcommon.h
 
+noinst_HEADERS  =       tpm_nvcommon.h
+
 noinst_PROGRAMS =	tpm_startup	\
 			tpm_reset
 
Only in tpm-tools-1.3.8/src/tpm_mgmt: Makefile.am.orig
